apps/experiments/views/experiment.py

```diff
@@ -641,11 +642,22 @@ def experiment_chat_session(request, team_slug: str, experiment_id: int, session
 @require_POST
 def experiment_session_message(request, team_slug: str, experiment_id: int, session_id: int):
     message_text = request.POST["message"]
+    uploaded_files = request.FILES
+    attachments = []
+    created_files = []
+    for file_type in ["code_interpreter", "file_search"]:
+        if file_type not in uploaded_files:
+            continue
+        for uploaded_file in uploaded_files.getlist(file_type):
+            new_file = File.objects.create(name=uploaded_file.name, file=uploaded_file, team=request.team)
```

Comment:
Why don't you create the `ThreadToolResources` object here too? There's some risk that the celery task fails and then these files will be orphaned. We're already fetching the session below so there's no additional query to get the chat ID (`session.chat_id`). It also saves you have to create them in the runnable.

Reply:
I was between creating it here and inside the assistants runnable. Happy to move it to the view :+1:

Reply:
https://github.com/dimagi/open-chat-studio/pull/525/commits/ba7e6c81462bdebfb60f6b64d37166a8b17a6a45